### PR TITLE
drivers: ethernet: cleanup Kconfig

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Review by erwango — DISMISSED** [review]

**Review by nandojve — DISMISSED** [review]

**Comment by galak** [comment]
Need CI fixes from: https://github.com/zephyrproject-rtos/zephyr/pull/48909 and https://github.com/zephyrproject-rtos/zephyr/pull/48908

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by ibirnbaum — DISMISSED** [review]

**Review by fkokosinski — DISMISSED** [review]

**Review by nandojve — DISMISSED** [review]

**Review by chrta — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

File: drivers/ethernet/CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,6 @@ zephyr_library_sources_ifdef(CONFIG_ETH_STELLARIS	eth_stellaris.c)
 zephyr_library_sources_ifdef(CONFIG_ETH_STM32_HAL	eth_stm32_hal.c)
 zephyr_library_sources_ifdef(CONFIG_ETH_W5500		eth_w5500.c)
 zephyr_library_sources_ifdef(CONFIG_ETH_SAM_GMAC	eth_sam_gmac.c)
-zephyr_library_sources_ifdef(CONFIG_DSA_KSZ8XXX		dsa_ksz8xxx.c)
```
Comment (mmahadevan108) [inline drivers/ethernet/CMakeLists.txt:30]: Why is this deleted?

Comment (henrikbrixandersen) [reply drivers/ethernet/CMakeLists.txt:30]: It's a duplicate. It's in the commit log.

**Comment by galak** [comment]
blocked on #48948

**Review by mmahadevan108 — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]
